### ISSUE TITLE
Add option for returning a default value in Map.get

### ISSUE THREAD

**Issue by gagdiez** [near/near-sdk-js#170]
I find that often `value = Map.get(key)` is followed by `value === null? some_default_value : value`. Maybe we could implement in the `get` a parameter for a default value to simplify this logic.

Similar to `unwrap_or` in Rust.

**Comment by petarvujovic98** [comment]
This could be as simple as doing the following:
```typescript
get(key: string, defaultValue?: T | null = null): T | null {
  // logic to retrieve the value
  if (value) {
    return value as T;
  } else if (defaultValue) {
    return defaultValue;
  } else {
    throw new Error(ERR_INCOSISTENT_STATE);
  }
}
```

**Comment by osalkanovic** [comment]
> This could be as simple as doing the following:
> 
> ```ts
> get(key: string, defaultValue?: T | null = null): T | null {
>   // logic to retrieve the value
>   if (value) {
>     return value as T;
>   } else if (defaultValue) {
>     return defaultValue;
>   } else {
>     throw new Error(ERR_INCOSISTENT_STATE);
>   }
> }
> ```

Yes, I thought about this, I would just add that the default value can be anything else, something like this: 
`
    get(key: Bytes): T | null;

    get<K extends unknown>(key: Bytes, defaultValue: K): T | K | null;

    get(key, defaultValue = null) {
        // logic to retrieve the value
        if (value) {
          return value as T;
        } else if (defaultValue) {
           return defaultValue;
        } else {
          throw new Error(ERR_INCOSISTENT_STATE);
        }
    }`

**Comment by petarvujovic98** [comment]
@osalkanovic I disagree with making the default value be _anything else_. This is going to add confusion and be bug prone. We want to reduce boilerplate and overhead for devs. 
I don't know of an instance where I would want to have a default value of a different type than the value I am trying to retrieve

**Comment by osalkanovic** [comment]
@petarvujovic98 
Got it, make sense.

**Comment by volovyks** [comment]
I like the design of @petarvujovic98.
@petarvujovic98 do you want to copy that and create a PR?

**Comment by petarvujovic98** [comment]
@volovyks I will create a PR

**Comment by petarvujovic98** [comment]
@volovyks This is also covered by the just merged #213

**Comment by ailisp** [comment]
@petarvujovic98 indeed. Thanks!
